### ISSUE TITLE
Center glyph vertically

### ISSUE THREAD

**Issue by abdelouahabb** [fontforge/fontforge#2227]
Hi,
Thank you for this wonderful software.
My problem is I can only select left/right guidlines (from the layers), I try to center this icon in the middle, but the lonly way I have now is to scale down.
In previous version, I remember I did that, but not anymore!
![sans titre](https://cloud.githubusercontent.com/assets/2527234/7358814/4558e398-ed30-11e4-9523-f4a7ba75f209.png)

**Comment by andreaslarsen** [comment]
I'm not completely sure what you're aiming to achieve from your description. You can set the width and center the glyph using the metrics menu
![screen shot 2015-04-28 at 00 06 58](https://cloud.githubusercontent.com/assets/3859425/7359033/833080ea-ed3a-11e4-8933-4a055db4463d.png)

**Comment by davelab6** [comment]
These are not guides, they are sideBearings :)

**Comment by abdelouahabb** [comment]
there is center in width but there is no center in height, which is the problem here :(

**Comment by andreaslarsen** [comment]
I don't believe that's been a feature i recent times and it's not really something you'd need for fonts. You can always export and edit in the vector editor you have (illustrator, inkscape etc.) - they are better suited for drawing icons anyway and then just import once everything is done. Just remember to use the same size in both programs [UPM](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=3&cad=rja&uact=8&ved=0CC0QFjAC&url=http%3A%2F%2Fdesignwithfontforge.com%2Fen-US%2FThe_EM_Square.html&ei=zG4_Vf-CFNXdavqVgYgN&usg=AFQjCNGtiEfWadozLaRclfW9nSyP7G8LFQ&sig2=vDM8-9ko6XJRMLRz1QI8sg&bvm=bv.91665533,d.d2s)

**Comment by abdelouahabb** [comment]
I ended up with scale down the chars, but I am pretty sure I was able to use the upper handler, but dont remember the version...

**Comment by andreaslarsen** [comment]
You're welcome:)

**Comment by abdelouahabb** [comment]
thank you again for this awesome software :+1: 

and please, add the option to change the language without deleting the locales, because the french translation is hell !

**Comment by davelab6** [comment]
The vertical metrics are

**Comment by andreaslarsen** [comment]
if/when someone creates such a script it should have an option to center in UPM and between baseline and ascent

**Comment by andreaslarsen** [comment]
and helping with translation is as simple as editing this file https://github.com/fontforge/fontforge/blob/master/po/fr.po

**Comment by abdelouahabb** [comment]
I wish I could, but I have first to find the right word, because it must be from calligraphy domain, which I am not :(

But seriously, I was really lost in the GUI because it was in french (I speak french), but as long as I deleted the po file, I got it clear!

**Comment by cpitclaudel** [comment]
This file seems to do something of this flavor: https://fontindustry.googlecode.com/files/centerglyph.py

**Comment by luizfilipe2557** [comment]
Nice @cpitclaudel , but I couldn't install this on FF, since I put the file on /python folder I don't know what to do anymore. Seems like my FF didn't recognize the script since the tools menu never changes. Could you help me?

**Comment by cpitclaudel** [comment]
Should be in the Fonts/Metrics and Fonts/Transform menus.

**Comment by luizfilipe2557** [comment]
Does not appear to be here :|

**Comment by luizfilipe2557** [comment]
In this places? 
![metrics](https://cloud.githubusercontent.com/assets/3942006/12094940/c03cc56a-b2f2-11e5-98eb-35523baf8256.PNG)
![transform](https://cloud.githubusercontent.com/assets/3942006/12094939/c0396afa-b2f2-11e5-9f91-22ecf77bc6aa.PNG)

Sorry if I'm being stupid, I'm a newbie in this area...

**Comment by cpitclaudel** [comment]
In that case I have no idea; I don't use that script. I guessed from this section:

```python

if fontforge.hasUserInterface():
    fontforge.registerMenuItem(GetSelectedBound, None, None, "Font", None,
            "Metrics", "Max Selected Size");
    fontforge.registerMenuItem(CenterHeight, None, None, "Font", None, "Metrics", 
            "Center in Height");
    fontforge.registerMenuItem(CenterGlyph, None, None, "Font", None, "Metrics", 
            "Center in Glyph");
    fontforge.registerMenuItem(ScaleToEm, None, None, "Font", None, "Transform", 
            "Scale to Em");
```

**Comment by davelab6** [comment]
All scripts menu items appear inside the tools menu; this sets up submenus
within it

**Comment by luizfilipe2557** [comment]
Yeah, but the code is from 2008 and FF is on a 2015 build. I think there's a conflict of versions...

**Comment by Yukinoroh** [comment]
Centering vertically is needed for Japanese fonts, because of vertical writing. A top side bearing and bottom side bearing would also be convenient, as I find the bounding box to be quite unreliable.

> I don't believe that's been a feature i recent times and it's not really something you'd need for fonts. You can always export and edit in the vector editor you have (illustrator, inkscape etc.) - they are better suited for drawing icons anyway and then just import once everything is done. Just remember to use the same size in both programs [UPM](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=3&cad=rja&uact=8&ved=0CC0QFjAC&url=http%3A%2F%2Fdesignwithfontforge.com%2Fen-US%2FThe_EM_Square.html&ei=zG4_Vf-CFNXdavqVgYgN&usg=AFQjCNGtiEfWadozLaRclfW9nSyP7G8LFQ&sig2=vDM8-9ko6XJRMLRz1QI8sg&bvm=bv.91665533,d.d2s)

**Comment by Francespo** [comment]
+1 on vertical centering

**Comment by unbeebnu** [comment]
#5824 
The issue is done.
